pkg/apis/pipeline/v1beta1/results.go

```diff
@@ -0,0 +1,26 @@
+/*
+Copyright 2022 The Tekton Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
```

Comment:
I see params for the similar file is called params_types.go. Maybe it worths being consistent with params?

Reply:
I have thought about this. 

1. Whether I should create a new file or put it into current file like task_types.go. I go with a new file for now because I think the data is used in different types (task, taskrun and maybe pipeline, pipelinerun in the future)
2. Naming of the new file, the files ended with "_types" should define new types in it but in our results.go we don't have new types. So for naming convention maybe without types makes more sense?

Reply:
Yes this is also something I'm not sure. I thought files ended with "_types" should have a crd in it.

Reply:
That's a pretty good consideration. I think TaskResult is actually similar to ParamsSpec: https://github.com/tektoncd/pipeline/blob/7e5e4c0f7b50c40192d46153580bfb1f38891a1b/pkg/apis/pipeline/v1beta1/param_types.go#L37 and TaskRunResult is similar to Param: https://github.com/tektoncd/pipeline/blob/7e5e4c0f7b50c40192d46153580bfb1f38891a1b/pkg/apis/pipeline/v1beta1/param_types.go#L83

Maybe we can move those structs into a centralized file results_types.go.

Reply:
yeah that would also work, I can do it and see how other ppl think of it